### ISSUE TITLE
Any plans to expand the scope to the tauri event system?

### ISSUE THREAD

**Issue by Zercerium** [tauri-apps/tauri-bindgen#180]
imo it would be nice to get also the [event system ](https://tauri.app/v1/guides/features/events/) typed. Any plans or ideas for that?

At the moment a possible solution is to use [specta](https://github.com/oscartbeaumont/specta) to create at least the typescript types automatically. But you can still misspell the event identifiers. I also rly like the idea of the `.wit` files.

Thanks for your great work!

**Comment by JonasKruckenberg** [comment]
Yeah so the way I plan to bring this to `tauri-bindgen` is through something completely different: `stream`s.
This is essentially the same as in GRPC if you're familiar with that, if not here's the gist:

```wit
interface log {
   func start_logging() -> stream<string>
}
```

which would translate to something like this for the host

```rust
trait Log {
   type StartLoggingStream = Pin<Box<dyn Stream<Item = String> + Send  + 'static>>;

   fn start_logging(&self) -> Result<Self:: StartLoggingStream>
}
```

Similar to how the `tonic` crate handles this.

On the guest side this would map to `AsyncIterator` for JS and `SomeSpecificStreamType<String>` for Rust probably.

So no _events_ in the classical tauri sense (imo superior though bc it forces cleaner code) but I hope it can fulfill all the use cases.
